### ISSUE TITLE
Frequency

### ISSUE THREAD

**Issue by saebekassebil** [danigb/pitch#1]
Hey Daniel,

So I was wondering. Again..

What do you think about a `pitch-fq` module, maybe like this:
```js
module.exports = function(coord, stdPitch) {
  if (typeof coord === 'number') {
    stdPitch = coord;
    return function(coord) {
      return stdPitch * Math.pow(2, (coord[0] * 12 + coord[1] * 7) / 12);
    }
  }

  stdPitch = stdPitch || 440;
  return stdPitch * Math.pow(2, (coord[0] * 12 + coord[1] * 7) / 12);
}
```

Should that be a separate module (my favourite), or should it be embedded here in `pitch`? And where do we draw the line between what's "right/minimal" and what's practical?

I'd appreciate any thoughts on this :)

**Comment by danigb** [comment]
Hi Jakob,

nice to read you again. I don't have an answer about what's right/minimal. I've read a lot about that lately, and I'm doing my own experiments. I think that could be a nice module. +1 for that one.

My biggest doubt is about what is the canonical form that best define a note. I like the mathematical simplicity of the coord system, but I dislike its obscure value. It's not easy (for me at least) to know what I'm talking about when I see [3, -2], what is the meaning of that. That's why I've started my own lib: what I decide is that the string 'd3' is the canonical form to represent a note. [note-parser](http://github.com/danigb/note-parser) is my first attempt to that.

Apart of this, I think the above module (you wrote) has functionality enough.

**Comment by saebekassebil** [comment]
[pitch-fq](https://github.com/saebekassebil/pitch-fq).

I started out with the canonical form, being a string as well. I did so, because I could think of no other way to represent a note. Then I learned about the coordinate approach, and liked it a lot, because it meant a lot less string-parsing, which to me has always seemed like an error-prone process.

I totally agree that the coordinate form is unsensible for normal human beings. But it's a kind of "absolute"/canonical form, that computers easily store and compute upon. I like this, because that lets me write a series of modules, meant for "human consumption" on top of this "rock solid" foundation.

I had a talk with @substack about this, and he thought, that the canonical form was simply the frequency of a tone. That is A4 would be 440, etc. I started out disliking this approach because of its "un-humane" feel, and a fear that I wouldn't be able to calculate if a tone was represented by `Fb` or `E` in music theory. He argued, that music theory was just something we humans had made up, and that the "core truth" was hidden in the frequency.

I now see, that it might be possible to calculate whether a given frequency is `Fb` or `E` by looking at context, which is in fact what we do when we analyse stuff with music theory.

Anyways, for now I'll stick to the coordinate system - just though I would share my thoughts on this :)
